src/transformers/models/qwen2_5_omni/modular_qwen2_5_omni.py

```diff
@@ -0,0 +1,4423 @@
+
+    def forward(self, x):
+        acts1, acts2 = self.activations[::2], self.activations[1::2]
+        for c1, c2, a1, a2 in zip(self.convs1, self.convs2, acts1, acts2):
+            xt = a1(x)
+            xt = c1(xt)
+            xt = a2(xt)
+            xt = c2(xt)
+            x = xt + x
+
+        return x
+
+
+@add_start_docstrings(
+    "The full Qwen2.5Omni Token2WavBigVGAN model. Which take mel spectrogram as input and predict waveform.",
+    QWEN2_5OMNI_START_DOCSTRING.format(config_class="Qwen2_5OmniBigVGANConfig"),
+)
+class Qwen2_5OmniToken2WavBigVGANModel(Qwen2_5OmniPreTrainedModel):
+    config_class = Qwen2_5OmniBigVGANConfig
+
+    def __init__(self, config: Qwen2_5OmniBigVGANConfig):
+        super().__init__(config)
+
+        self.num_kernels = len(config.resblock_kernel_sizes)
+        self.num_upsamples = len(config.upsample_rates)
+
+        # pre conv
+        self.conv_pre = nn.Conv1d(config.mel_dim, config.upsample_initial_channel, 7, 1, padding=3)
+
+        # transposed conv-based upsamplers. does not apply anti-aliasing
+        self.ups = nn.ModuleList()
+        for i, (u, k) in enumerate(zip(config.upsample_rates, config.upsample_kernel_sizes)):
+            self.ups.append(
+                nn.ModuleList(
+                    [
+                        ConvTranspose1d(
+                            config.upsample_initial_channel // (2**i),
+                            config.upsample_initial_channel // (2 ** (i + 1)),
+                            k,
+                            u,
+                            padding=(k - u) // 2,
+                        )
+                    ]
+                )
+            )
+
+        # residual blocks using anti-aliased multi-periodicity composition modules (AMP)
+        self.resblocks = nn.ModuleList()
+        for i in range(len(self.ups)):
+            ch = config.upsample_initial_channel // (2 ** (i + 1))
+            for j, (k, d) in enumerate(zip(config.resblock_kernel_sizes, config.resblock_dilation_sizes)):
+                self.resblocks.append(AMPBlock(ch, k, d))
+
+        # post conv
+        self.activation_post = TorchActivation1d(activation=SnakeBeta(ch))
+
+        self.conv_post = nn.Conv1d(ch, 1, 7, 1, padding=3, bias=False)
+
+    def _normalize(self, S, max_abs_value, min_db):
+        return torch.clamp(
+            (2 * max_abs_value) * ((S - min_db) / (-min_db)) - max_abs_value, -max_abs_value, max_abs_value
+        )
+
+    def _amp_to_db(self, x, min_level_db):
+        min_level = np.exp(min_level_db / 20 * np.log(10))
+        min_level = torch.ones_like(x) * min_level
+        return 20 * torch.log10(torch.maximum(min_level, x))
+
+    def apm_to_db(self, apm_mel):
+        mel_spec = torch.exp(apm_mel)
+
+        mel_spec = self._amp_to_db(mel_spec, -115) - 20
+        mel_spec = self._normalize(mel_spec, 1, -115)
+
+        return mel_spec
+
+    def forward(self, apm_mel):
+        mel_spec = self.apm_to_db(apm_mel)
```

Comment:
the audio processor should take care of this!

Reply:
Here the `apm_mel` is the output of dit module, we can't get the value in audio processor.